compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1600,13 +1660,23 @@ impl Type {
         let target = target.follow_bindings();
 
         if let (Type::Array(_size, element1), Type::Slice(element2)) = (&this, &target) {
-            // Still have to ensure the element types match.
-            // Don't need to issue an error here if not, it will be done in unify_with_coercions
-            let mut bindings = TypeBindings::new();
-            if element1.try_unify(element2, &mut bindings).is_ok() {
-                convert_array_expression_to_slice(expression, this, target, interner);
-                Self::apply_type_bindings(bindings);
-                return true;
+            // We can only do the coercion if the `as_slice` method exists.
+            // This is usually true, but some tests don't have access to the standard library.
+            if let Some(as_slice_method) = interner.lookup_primitive_method(&this, "as_slice") {
+                // Still have to ensure the element types match.
+                // Don't need to issue an error here if not, it will be done in unify_with_coercions
+                let mut bindings = TypeBindings::new();
+                if element1.try_unify(element2, &mut bindings).is_ok() {
+                    convert_array_expression_to_slice(
+                        expression,
+                        this,
+                        target,
+                        as_slice_method,
+                        interner,
+                    );
```

Comment:
```suggestion
            if let Some(as_slice) = interner.lookup_primitive_method(&this, "as_slice") {
                // Still have to ensure the element types match.
                // Don't need to issue an error here if not, it will be done in unify_with_coercions
                let mut bindings = TypeBindings::new();
                if element1.try_unify(element2, &mut bindings).is_ok() {
                    convert_array_expression_to_slice(expression, this, target, as_slice, interner);
```
Renaming to `as_slice` lets this fit within the 100 column limit